### ISSUE TITLE
Feat: Custom URL Schema Handlers not only app Protocol Handlers

### ISSUE THREAD

**Issue by lemanschik** [nwjs/nw.js#8143]
### Issue Type

Before opening an issue, please search and see if it has already been raised.

- [ ] Bug Report
- [x] Feature Request

- [x] Successfully reproduced against the latest version of NW.js?

> Please use our [mailing list](https://groups.google.com/forum/#!forum/nwjs-general) or [Gitter chatroom](https://gitter.im/nwjs/nw.js) to ask questions. The issue tracker is only for bugs and feature requests, in English only. Please note that issues without a repro or code snippet are less likely to be resolved.

### Current/Missing Behavior
Electron like custom url handler registration

### Expected/Proposed Behavior
Lets add the patches from brave to implement custom url handling fast.
https://github.com/brave/brave-browser/wiki/Adding-a-protocol-scheme-to-Brave

### Additional Info

- Operating System:
- NW.js Version:
- Repro Link:
- Code snippet:
- Crash report:

**Comment by ayushmanchhabra** [comment]
https://github.com/Shubham-Kumar-2000/protocol-registry ?

**Comment by lemanschik** [comment]
@ayushmanchhabra thats correct that looks at last ok in the 5 min that i reviewed it but i am not sure if it Handels all supported Platforms eg: MacOS, Windows and so on but general ya you got the right idea.

But you did not fully read and understand the issue there are diffrent types of protocol handlers

and the most importent type is not integrate able client side in js. the BiDirectional HTTP Like stream thats why i choosed the Issue text from brave to make that clear.
